src/Tactic/Cong.agda

```diff
@@ -0,0 +1,185 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Macros for making working with cong more convenient
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Level renaming (suc to lsuc; _⊔_ to lmax)
+open import Relation.Binary.Definitions
+open import Tactic.Cong.Id
+open import Data.Nat renaming (_≟_ to _≟ℕ_)
+
+module Tactic.Cong
+  (recursion-limit : ℕ)
+  (_≡_ : ∀ {ℓ} {A : Set ℓ} → A → A → Set ℓ)
```

Comment:
As this is an arbitrary equality can we use `\approx` instead of `\equiv`? Otherwise it's a bit confusing that it's not propositional equality. Hmm actually this type is interesting. You have to provide an equality over every type? So actually it does only work for propositional equality?

Reply:
We also tend to use `a` and `b` for set levels, and `ℓ` is reserved for levels for the `Rel` type

Reply:
`\approx` makes sense.

Regarding the type signature, that's an oversight on my part. I'll have to see what I can do to make it take both relations that are generic over types and those that are not. Are there any existing signatures for this sort of thing in the library? Perhaps for relations over families of types?

Regarding the use of `ℓ`, that's accidental use of conventions from https://github.com/agda/cubical on my part. I'll change it.

Reply:
This is the signature for the module I am trying to get to work now:
```agda
module Tactic.Cong
  (recursion-limit : ℕ)
  (Param : Setω)
  (Family-Level : Param → Level)
  (Family : (A : Param) → Set (Family-Level A))
  (_<_ : ∀ {A : Param} → Family A → Family A → Set (Family-Level A))
  (trans : ∀ {A : Param} → Transitive {A = Family A} _<_)
  (_≈_ : ∀ {a} {A : Set a} → A → A → Set a)
  (cong : {a b : Level}
          {A : Set a}
          {B : Set b}
          (f : A → B)
          {x y : A}
          (p : x ≈ y)
          → (f x) ≈ (f y))
```

Reply:
It is proving a bit difficult to get this signature to work, mainly due to https://github.com/agda/agda/issues/4576

One possible way of fixing it is to change the signature of the `trans` parameter to `Term`. A user would then have to pass `quoteTerm trans` instead of `trans`. This then misses out on type checking of `trans` when the module is imported.

Reply:
It actually looks like fixing this properly would only require a very small patch to Agda.

Reply:
I have now opened https://github.com/agda/agda/pull/4579. If that PR goes through, I guess this PR would need to be made against the experimental branch instead of master?

Reply:
Resolving, since https://github.com/agda/agda/pull/4579 has been merged.